src/tools/run-make-support/src/clang.rs

```diff
@@ -0,0 +1,77 @@
+use std::env;
+use std::path::Path;
+use std::process::Command;
+
+use crate::{bin_name, handle_failed_output, tmp_dir};
+
+/// Construct a new `clang` invocation. `clang` is not always available for all targets, you
+/// should check if you need `//@ needs-matching-clang` to make sure `clang` is available for
```

Comment:
If we want to refactor this into a common utility, then some tests might only need clang but don't need a matching clang.

Matching clang is needed for cross-lang LTO.

Reply:
Ah good point, I'll edit the comment. Thanks.

Reply:
I removed the `needs-matching-clang` comment, because when `clang` is available is quite complicated (I don't actually know precisely when `clang` is available and when it's not). If someone knows, I would be happy to update this comment.